<!DOCTYPE html>
<html lang="en"><head>
<script src="test_2_082025_files/libs/clipboard/clipboard.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/tabby.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/popper.min.js"></script>
<script src="test_2_082025_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="test_2_082025_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="test_2_082025_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="test_2_082025_files/libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.33">

  <meta name="author" content=" Lesley Chapman Hannah, Ph.D., M.S. College of Graduate Studies Northeast Ohio Medical University">
  <title>Practical Regression Implementations: Cox, Poisson, and Negative Binomial Models</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="test_2_082025_files/libs/revealjs/dist/theme/quarto-2c1b5f745a11cfad616ebade4a4a7d24.css">
  <link href="test_2_082025_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="test_2_082025_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h2 style="color: #2A76DD;" class="title">Practical Regression Implementations: Cox, Poisson, and Negative Binomial Models</h2>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div style="font-size:20px;" class="quarto-title-author-name">
<br><br><span style="font-size:25px;"><strong>Lesley Chapman Hannah, Ph.D., M.S.</strong></span><br>College of Graduate Studies<br>Northeast Ohio Medical University 
</div>
</div>
</div>

</section>
<section id="overview" class="slide level2">
<h2>Overview</h2>
<div style="font-size:0.8em">
<ul>
<li><p>This lecture extends the regression framework to new outcome types using the same structure as previously discussed regression models</p></li>
<li><p>Each model answers a different biomedical question by changing the outcome distribution</p></li>
</ul>
</div>
</section>
<section>
<section id="poisson-regression-count-outcomes" class="title-slide slide level1 center">
<h1>Poisson Regression : Count Outcomes</h2>

</section>
<section id="poisson-regression-research-questions" class="slide level2">
<h2>Poisson Regression — Research Question(s)</h2>
<p>Do patients with chronic obstructive pulmonary disease (COPD)<br>
experience more acute exacerbations [sudden worsening of respiratory symptoms] per year as baseline lung function (FEV1) decreases?</p>
<ul>
<li>Outcome: number of exacerbations in 12 months</li>
<li>Predictors: FEV1 [Forced Expiratory Volume in 1 second], smoking history</li>
</ul>
</section>
<section id="poisson-regression-model-statement" class="slide level2">
<h2>Poisson Regression — Model Statement</h2>
<div style="font-size:0.8em">
<p><span class="math display">\[
Y_i \sim \text{Poisson}(\lambda_i), \quad
\log(\lambda_i) = \beta_0 + \beta_1 \text{FEV1}_i + \beta_2 \text{Smoking}_i
\]</span></p>
<ul>
<li><span class="math inline">\(\lambda_i\)</span>: expected number of exacerbations</li>
</ul>
</div>
</section>
<section id="poisson-distribution" class="slide level2">
<h2>Poisson Distribution</h2>
<div style="font-size:0.8em">
<p>Poisson likelihood assumes:</p>
<ul>
<li>discrete counts (0,1,2,…)</li>
<li>variance approximately equal to the mean</li>
</ul>
<p><span class="math display">\[
Yi​∼Poisson(λi​)
\]</span></p>
<p>Assumptions for each patient:</p>
<ul>
<li><span class="math inline">\(Y_i\)</span> is a count (0, 1, 2, etc.)</li>
<li>events occur independently</li>
<li>expected number of events is: <span class="math inline">\(λ_i\)</span></li>
<li>in the Poisson distribution the mean equals variance <span class="math display">\[
E(Yi​)=Var(Yi​)=λi​
\]</span></li>
</ul>
</div>
</section>
<section id="poisson-distribution-1" class="slide level2">
<h2>Poisson Distribution</h2>
<div style="font-size:0.5em">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.006</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.041368</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-2-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="poisson-distribution-2" class="slide level2">
<h2>Poisson Distribution</h2>
<div style="font-size:0.8em">
<ul>
<li>average count and variance = 2</li>
<li>most patients have 0–3 events</li>
<li>fewer patients have large counts</li>
<li>distribution is right-skewed</li>
<li>counts are discrete, not continuous</li>
</ul>
</div>
</section>
<section id="simulated-dataset" class="slide level2">
<h2>Simulated Dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a><span class="fu">head</span>(dat_pois)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
   fev1 smoking exacerbations
  &lt;dbl&gt;   &lt;int&gt;         &lt;int&gt;
1  34.5       1             4
2  77.4       1             3
3  81.5       0             2
4  37.8       0             8
5  33.3       1             8
6  39.2       1             8</code></pre>
</div>
</div>
</section>
<section id="poisson-regression-model-fit" class="slide level2">
<h2>Poisson Regression — Model Fit</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a>fit_pois <span class="ot">&lt;-</span> <span class="fu">glm</span>(exacerbations <span class="sc">~</span> fev1 <span class="sc">+</span> smoking,</span>
<span id="cb5-2"><a href=""></a>                <span class="at">family =</span> <span class="fu">poisson</span>(),</span>
<span id="cb5-3"><a href=""></a>                <span class="at">data =</span> dat_pois)</span>
<span id="cb5-4"><a href=""></a><span class="fu">summary</span>(fit_pois)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = exacerbations ~ fev1 + smoking, family = poisson(), 
    data = dat_pois)

Coefficients:
             Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)  2.666410   0.122220  21.817  &lt; 2e-16 ***
fev1        -0.029431   0.001918 -15.344  &lt; 2e-16 ***
smoking      0.341847   0.075789   4.511 6.47e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 550.91  on 249  degrees of freedom
Residual deviance: 252.99  on 247  degrees of freedom
AIC: 905.32

Number of Fisher Scoring iterations: 5</code></pre>
</div>
</div>
</section>
<section id="interpreting-the-coefficients-table" class="slide level2">
<h2>Interpreting the Coefficients Table</h2>
<div style="font-size:0.8em">
<p>modeled the number of acute COPD exacerbations over 12 months as a function of:</p>
<ul>
<li>FEV1 (% predicted): baseline lung function</li>
<li>smoking status: smoker vs non-smoker</li>
</ul>
<p>Coefficients Table</p>
<ul>
<li><strong>Estimate</strong>: change in <em>log</em> expected count</li>
<li><strong>exp(Estimate)</strong>: effect on the rate for a 1-unit change in <span class="math inline">\(X_i\)</span> [i.e.: exacerbation rate]</li>
<li><strong>Std. Error</strong>: uncertainty of the estimate</li>
</ul>
</div>
</section>
<section id="interpreting-the-coefficients-table-1" class="slide level2">
<h2>Interpreting the Coefficients Table</h2>
<div style="font-size:0.7em">
<p>Coefficients Table Examples:</p>
<p>Intercept : 2.666</p>
<ul>
<li>exp(2.666)≈14.4</li>
<li>14.4 exacerbations per year for a reference patient:</li>
<li>FEV1 = 0</li>
<li>non-smoker</li>
<li>intercept anchors the model and is rarely interpreted</li>
</ul>
</div>
</section>
<section id="interpreting-the-coefficients-table-2" class="slide level2">
<h2>Interpreting the Coefficients Table</h2>
<div style="font-size:0.7em">
<p>Coefficients Table Examples:</p>
<p>FEV1: −0.029</p>
<ul>
<li>exp(−0.029)≈0.97</li>
<li>each 1% increase in FEV1 was associated with a 3% reduction in expected exacerbations</li>
</ul>
<p>smoking status: 0.342</p>
<ul>
<li>exp(0.342)≈1.41</li>
<li>smokers experience about 41% more acute exacerbations per year than non-smokers</li>
</ul>
<p>Statistical Significance:</p>
<ul>
<li>both FEV1 and smoking have very strong statistical support</li>
<li>p-values &lt; 0.001 indicate the observed effects are unlikely due to chance under the Poisson model</li>
</ul>
</div>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="negative-binomial-regression-overdispersed-counts" class="title-slide slide level1 center">
<h1>Negative Binomial Regression [Overdispersed Counts]</h2>

</section>
<section id="negative-binomial-research-questions" class="slide level2">
<h2>Negative Binomial — Research Question(s)</h2>
<p>Among RNA-seq samples, does sequencing depth predict the number of detected low-abundance transcripts?</p>
<ul>
<li>Outcome: number of low-expression genes detected</li>
<li>Predictors: sequencing depth, RNA integrity score</li>
</ul>
</section>
<section id="negative-binomial-model-statement" class="slide level2">
<h2>Negative Binomial — Model Statement</h2>
<div style="font-size:0.8em">
<p><span class="math display">\[
Y_i \sim \text{NB}(\mu_i, \theta), \quad
\log(\mu_i) = \beta_0 + \beta_1 \text{Depth}_i + \beta_2 \text{RIN}_i
\]</span></p>
<ul>
<li><span class="math inline">\(\mu_i\)</span>: expected count
<ul>
<li>average number of low-abundance transcripts we expect to detect for sample <em>i</em></li>
</ul></li>
<li><span class="math inline">\(\theta\)</span>: dispersion parameter controlling extra variability around the mean of each sample</li>
</ul>
</div>
</section>
<section id="negative-binomial-distribution" class="slide level2">
<h2>Negative Binomial Distribution</h2>
<div style="font-size:0.8em">
<p>Negative binomial distribution:</p>
<p><span class="math display">\[
Y_i∼NB(μ_i,θ)
\]</span></p>
<ul>
<li><span class="math inline">\(Y_i\)</span>: count (0, 1, 2, etc)</li>
<li><span class="math inline">\(μ_i\)</span> : expected count</li>
<li>θ : dispersion parameter that allows more spread than Poisson</li>
<li>Negative binomial: variance exceeds mean unlike Poisson where variance = mean</li>
</ul>
</div>
</section>
<section id="negative-binomial-distribution-1" class="slide level2">
<h2>Negative Binomial Distribution</h2>
<div style="font-size:0.8em">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 3
  model              mean    var
  &lt;chr&gt;             &lt;dbl&gt;  &lt;dbl&gt;
1 NegBin (theta=1)  10.0  112.  
2 NegBin (theta=50) 10.0   11.9 
3 Poisson            9.93   9.80</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="test_2_082025_files/figure-revealjs/unnamed-chunk-6-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="negative-binomial-distribution-2" class="slide level2">
<h2>Negative Binomial Distribution</h2>
<div style="font-size:0.8em">
<ul>
<li>three models :
<ul>
<li>Negative Binomial with <span class="math inline">\(\theta\)</span> = 1 and 50</li>
<li>Poisson model</li>
</ul></li>
<li>mean ~10 events per model</li>
<li>NB <span class="math inline">\(\theta\)</span> = 1 : counts vary widely
<ul>
<li>samples vary widely and is representative of whats observed in studies</li>
</ul></li>
<li>NB <span class="math inline">\(\theta\)</span> = 50 : variance [11.9] is closer to the mean
<ul>
<li>as θ becomes large <span class="math inline">\(\rightarrow\)</span> negative binomial behaves more and more like Poisson</li>
</ul></li>
</ul>
</div>
</section>
<section id="simulated-dataset-1" class="slide level2">
<h2>Simulated Dataset</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  depth   rin genes_lo_exp
  &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt;
1  13.7  7.39          113
2  12.2  7.11           21
3  39.9  6.72           65
4  24.5  8.00          156
5  29.8  6.03           22
6  35.7  9.08           10</code></pre>
</div>
</div>
</section>
<section id="negative-binomial-model" class="slide level2">
<h2>Negative Binomial Model</h2>
<div style="font-size:0.8em">
<ul>
<li><p>modeled the number of low-expression genes detected in an RNA-seq sample as a function of:</p>
<ul>
<li>sequencing depth (e.g., millions of reads)</li>
<li>RNA integrity number (RIN)</li>
</ul></li>
</ul>
</div>
</section>
<section id="negative-binomial-model-fit" class="slide level2">
<h2>Negative Binomial — Model Fit</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href=""></a>fit_nb <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(genes_lo_exp <span class="sc">~</span> depth <span class="sc">+</span> rin, <span class="at">data =</span> dat_nb)</span>
<span id="cb9-2"><a href=""></a><span class="fu">summary</span>(fit_nb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm.nb(formula = genes_lo_exp ~ depth + rin, data = dat_nb, init.theta = 1.112880131, 
    link = log)

Coefficients:
            Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept) 3.196920   0.451902   7.074 1.50e-12 ***
depth       0.018476   0.004301   4.296 1.74e-05 ***
rin         0.191712   0.052461   3.654 0.000258 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for Negative Binomial(1.1129) family taken to be 1)

    Null deviance: 304.73  on 239  degrees of freedom
Residual deviance: 273.07  on 237  degrees of freedom
AIC: 3063.4

Number of Fisher Scoring iterations: 1

              Theta:  1.1129 
          Std. Err.:  0.0917 

 2 x log-likelihood:  -3055.3930 </code></pre>
</div>
</div>
</section>
<section id="negative-binomial-interpreting-the-coefficients-table" class="slide level2">
<h2>Negative Binomial — Interpreting the Coefficients Table</h2>
<div style="font-size:0.8em">
<p><strong>Intercept</strong>: 3.197 - exp(3.197)≈24.4 - model predicts about 24 low-expression genes detected for a reference sample with: - depth = 0 - RIN = 0</p>
<p><strong>sequencing depth</strong>: 0.0185 - exp(0.0185)≈1.019 - each 1-unit increase in sequencing depth is associated with a 1.9% increase in the expected number of detected low-expression genes - essentially, higher sequencing depth leads to greater sensitivity of detecting low-abundance transcripts</p>
<p><strong>RNA integrity number</strong>: 0.192 - exp(0.192)≈1.21 - each 1-unit increase in RIN is associated with a 21% increase in the expected number of detected low-expression genes - essentially, higher RMA omtegreity reseves fragile transcripts</p>
</div>
</section>
<section id="negative-binomial-interpreting-the-coefficients-table-1" class="slide level2">
<h2>Negative Binomial — Interpreting the Coefficients Table</h2>
<div style="font-size:0.8em">
<p><strong>Statistical Significance</strong>: Both depth and RIN have strong statistical support (p &lt; 0.001)</p>
<p><strong>Dispersion Parameter</strong>: - θ≈1.1 - variance is much larger than the mean - model overdispersion - Poisson model would underestimate uncertainty</p>
</div>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="cox-regression-time-to-event-outcomes" class="title-slide slide level1 center">
<h1>Cox Regression : Time-to-Event Outcomes</h2>

</section>
<section id="cox-regression-research-questions" class="slide level2">
<h2>Cox Regression — Research Question(s)</h2>
<p>Among patients with early-stage breast cancer, does higher tumor proliferation predict shorter time to seeing disease manifest in a patient?</p>
<ul>
<li>outcome: time to recurrence</li>
<li>predictors: Ki-67 index, age</li>
</ul>
<p>Definition(s): - <strong>recurrence</strong>: first time an event of interest occurs after a defined starting point - first confirmed return of breast cancer after initial curative treatment - Ki-67: tumor proliferation will be defined by presence of Ki-67 a tumor proliferation marker</p>
</section>
<section id="cox-regression-model-statement" class="slide level2">
<h2>Cox Regression — Model Statement</h2>
<div style="font-size:0.7em">
<p><span class="math display">\[
h(t \mid X_i) = h_0(t) \exp(\beta_1 \text{Ki67}_i + \beta_2 \text{Age}_i)
\]</span></p>
<p>Key parts in context:</p>
<ul>
<li><span class="math inline">\(h(t∣X_i)\)</span>: hazard - rate of an event at a set time</li>
<li><span class="math inline">\(h(t)\)</span>: risk of recurrence at time <span class="math inline">\(t\)</span></li>
<li><span class="math inline">\(h_0(t)\)</span>: baseline recurrence risk over time
<ul>
<li>recurrence risk over time for a reference point</li>
</ul></li>
<li><span class="math inline">\(\beta_1\)</span>: log hazard ratio per unit increase in Ki-67
<ul>
<li>how tumor proliferation changes recurrence risk</li>
</ul></li>
<li><span class="math inline">\(\beta_2\)</span>: log hazard ratio per year of age
<ul>
<li>how recurrence risk changes per additional year of age</li>
</ul></li>
</ul>
</div>
</section>
<section id="cox-regression-likelihood" class="slide level2">
<h2>Cox Regression — Likelihood</h2>
<div style="font-size:0.6em">
<p>Cox model is fit using a <strong>partial likelihood</strong>:</p>
<ul>
<li>helps answer: does adding this set of predictors (e.g., Ki-67) improve the model’s explanation of the observed recurrence times by comparing two models:
<ul>
<li>reduced model : model with age only as a predictor</li>
<li>full model: model with age and Ki-67</li>
</ul></li>
<li>likelihood ratio compares how well each model explains the data:</li>
</ul>
<p><span class="math display">\[
\text{Likelihood ratio} = \frac{L(\text{full model})}{L(\text{reduced model})}
\]</span></p>
</div>
</section>
<section id="cox-regression-likelihood-1" class="slide level2">
<h2>Cox Regression — Likelihood</h2>
<div style="font-size:0.6em">
<p>Coefficients are interpreted via <strong>hazard ratios</strong>:</p>
<p><span class="math display">\[
HR = \exp(\beta)
\]</span></p>
<ul>
<li>HR &gt; 1: higher recurrence risk</li>
<li>HR &lt; 1: lower recurrence risk</li>
</ul>
</div>
</section>
<section id="simulated-dataset-2" class="slide level2">
<h2>Simulated dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href=""></a><span class="fu">head</span>(dat_cox)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 4
    age  ki67  time status
  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;int&gt;
1  63.6 15.0   4.18      0
2  77.6 20.0   1.36      0
3  39.6 16.8  87.9       1
4  36.8 18.0  10.2       1
5  55.9  8.36 11.6       1
6  38.8 15.3  73.3       0</code></pre>
</div>
</div>
</section>
<section id="cox-regression-model" class="slide level2">
<h2>Cox Regression Model</h2>
<div style="font-size:0.8em">
<ul>
<li>modeling time to first breast cancer recurrence as a function of: tumor proliferation [Ki-67] and age</li>
<li>evaluate patients’ current risk of recurrence over time</li>
<li>Surv(time, status) :
<ul>
<li>among patients who have not yet recurred, who is more likely to recur next?</li>
<li>factors that define the survival outcome:
<ul>
<li>time: how long the patient was followed</li>
<li>status: recurrence[1] vs no recurrence [0]</li>
</ul></li>
</ul></li>
</ul>
</div>
</section>
<section id="cox-regression-model-fit" class="slide level2">
<h2>Cox Regression — Model Fit</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href=""></a>fit_cox <span class="ot">&lt;-</span> <span class="fu">coxph</span>(<span class="fu">Surv</span>(time, status) <span class="sc">~</span> ki67 <span class="sc">+</span> age, <span class="at">data =</span> dat_cox)</span>
<span id="cb13-2"><a href=""></a><span class="fu">summary</span>(fit_cox)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call:
coxph(formula = Surv(time, status) ~ ki67 + age, data = dat_cox)

  n= 220, number of events= 140 

        coef exp(coef) se(coef)     z Pr(&gt;|z|)    
ki67 0.05730   1.05897  0.01225 4.676 2.92e-06 ***
age  0.01803   1.01820  0.00682 2.645  0.00818 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

     exp(coef) exp(-coef) lower .95 upper .95
ki67     1.059     0.9443     1.034     1.085
age      1.018     0.9821     1.005     1.032

Concordance= 0.607  (se = 0.027 )
Likelihood ratio test= 24.26  on 2 df,   p=5e-06
Wald test            = 27.16  on 2 df,   p=1e-06
Score (logrank) test = 27.39  on 2 df,   p=1e-06</code></pre>
</div>
</div>
</section>
<section id="cox-regression-interpreting-the-coefficients-table" class="slide level2">
<h2>Cox Regression — Interpreting the Coefficients Table</h2>
<div style="font-size:0.6em">
<p>rows represent covariate-adjusted effect on recurrence risk:</p>
<ul>
<li><strong>coef</strong>: effect on the log hazard scale</li>
<li><strong>exp(coef)</strong>: hazard ratio</li>
<li><strong>se(coef)</strong>: uncertainty in the estimated effect</li>
<li><strong>p-value</strong>: evidence that the hazard ratio differs from 1</li>
</ul>
<p><strong>Ki-67 hazard ratio</strong>:</p>
<ul>
<li>exp(coef) = 1.033</li>
<li>given : patients who are the same age recurrence-free at time 0</li>
<li>patient with a higher Ki-67 value [i.e.: 1 unit higher] has a 3.3% higher current risk of recurrence</li>
</ul>
<p><strong>Age hazard ratio</strong>:</p>
<ul>
<li>exp(coef) = 1.015</li>
<li>an additional year of age confers a 1.5% higher current risk of recurrence</li>
</ul>
</div>
</section>
<section id="cox-regression-interpreting-the-coefficients-table-1" class="slide level2">
<h2>Cox Regression — Interpreting the Coefficients Table</h2>
<div style="font-size:0.6em">
<p><strong>95% Confidence Intervals</strong></p>
<ul>
<li>95% confident that each 1-unit increase in Ki-67 increases the recurrence risk by between 1.3% and 5.4%</li>
<li>95% confident that each additional year of age increases current recurrence risk by between 0.1% and 2.9%</li>
</ul>
</div>
</section>
<section class="slide level2">


</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="test_2_082025_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="test_2_082025_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="test_2_082025_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="test_2_082025_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>