<!DOCTYPE html>
<html lang="en"><head>
<script src="test_2_082025_AIC_BIC_updated_files/libs/clipboard/clipboard.min.js"></script>
<script src="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/tabby.min.js"></script>
<script src="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/popper.min.js"></script>
<script src="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="test_2_082025_AIC_BIC_updated_files/libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.33">

  <meta name="author" content=" Lesley Chapman Hannah, Ph.D., M.S. College of Graduate Studies Northeast Ohio Medical University">
  <title>Common Structure Behind Regression</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/dist/theme/quarto-2c1b5f745a11cfad616ebade4a4a7d24.css">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Common Structure Behind Regression</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
<br><br><span style="font-size:25px;"><strong>Lesley Chapman Hannah, Ph.D., M.S.</strong></span><br>College of Graduate Studies<br>Northeast Ohio Medical University 
</div>
</div>
</div>

</section>
<section id="regression-review" class="slide level2">
<h2>Regression Review</h2>
<ul>
<li>in biomedical research questions often involve understanding how biological or clinical variables relate to an outcome of interest [i.e.: treatment response]</li>
<li>substantial variability arises from a number of sources [i.e.: technical noise and unmeasured factors]</li>
<li>statistical models such as linear regression must simultaneously describe systematic effects while accounting for uncertainty</li>
<li>linear regression provides a concise framework that accomplishes this</li>
</ul>
</section>
<section id="linear-regression-model-overview" class="slide level2">
<h2>Linear regression model Overview</h2>
<div style="font-size:0.8em">
<p>Linear regression model</p>
<p><span class="math display">\[
Y=Xβ+ε,ε∼N(0,σ2)
\]</span></p>
<ul>
<li><span class="math inline">\(Xβ\)</span> captures the explainable relationship between predictors X and outcome Y</li>
<li>β represents effect sizes (how much Y changes when X changes)</li>
<li>explained in more detail in a later slide</li>
</ul>
</div>
</section>
<section id="linear-regression-model-overview-1" class="slide level2">
<h2>Linear regression model Overview</h2>
<div style="font-size:0.8em">
<p>Linear regression model</p>
<p><span class="math display">\[
Y=Xβ+ε,ε∼N(0,σ2)
\]</span></p>
<p>Randomness and uncertainty</p>
<p><span class="math display">\[
ε∼N(0,σ2)
\]</span></p>
<ul>
<li>part of the model that accounts for noisiness in data</li>
<li>ε represents unobserved factors, measurement error, and randomness</li>
<li>distributional effect allows one to quantify uncertainty</li>
</ul>
</div>
</section>
<section id="linear-regression-model-overview-2" class="slide level2">
<h2>Linear regression model Overview</h2>
<div style="font-size:0.8em">
<ul>
<li>Model can be written as a probability which helps one determine how likely different outcomes are</li>
</ul>
<p><span class="math display">\[
Y∣X∼N(Xβ,σ2)
\]</span></p>
<ul>
<li>likelihood is a probability distribution as a function of the parameters not the data</li>
<li>measures how plausible the observed data are for different β values</li>
</ul>
<p><span class="math display">\[
p(Y∣X,β, σ^2)
\]</span></p>
</div>
</section>
<section id="linear-regression-and-sampling" class="slide level2">
<h2>Linear regression and sampling</h2>
<div style="font-size:0.8em">
<ul>
<li>regression models how the observed outcomes Y arise from random draws from a population given the predictors X</li>
<li>Each observation is treated as a random draw from a Normal (Gaussian) distribution with mean <span class="math inline">\(X_i\)</span>β and variance <span class="math inline">\(σ^2\)</span></li>
<li>Each observation is essentially a sample from a probability distribution:</li>
</ul>
<p><span class="math display">\[
Yi​∣Xi​∼N(Xi​β,σ2)
\]</span></p>
<ul>
<li>β captures the systematic effect of predictors</li>
<li><span class="math inline">\(σ^2\)</span> represents the inherent variability</li>
</ul>
</div>
</section>
<section id="linear-regression-and-inference" class="slide level2">
<h2>Linear regression and inference</h2>
<div style="font-size:0.7em">
<ul>
<li>inference is the process of using data from a sample to make conclusions about the underlying population or process that generated the data</li>
<li><code>confidence intervals (CIs)</code> :
<ul>
<li>provide a range of plausible values for value being estimated [i.e.: <span class="math inline">\(\beta\)</span>]</li>
<li>e.x.: if a model provides an estimate with a 95% CI this means within the context of sampling - if we repeated the experiment many times, 95% of the resulting intervals would contain the true we’re attempting to estimate [i.e.: <span class="math inline">\(\beta\)</span>]</li>
</ul></li>
<li><code>p-values</code> :
<ul>
<li>test to compare observed values <span class="math inline">\(\hat{\beta}\)</span> to its expected distribution under a null hypothesis [i.e.:β=0]</li>
<li>small p-value → evidence against the null hypothesis</li>
<li>large p-value → not enough evidence to reject the null hypothesis</li>
</ul></li>
</ul>
</div>
</section>
<section id="linear-regression-and-inference-1" class="slide level2">
<h2>Linear regression and inference</h2>
<div style="font-size:0.4em">
<table class="caption-top">
<colgroup>
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 16%">
<col style="width: 64%">
</colgroup>
<thead>
<tr class="header">
<th><strong>Model</strong></th>
<th><strong>Outcome Type</strong></th>
<th><strong>Likelihood / Distribution</strong></th>
<th><strong>Inference (What we can test)</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Linear Regression</strong></td>
<td>Continuous (Y)</td>
<td>Gaussian</td>
<td>t-tests <br> confidence intervals <br> F-tests for overall model <br> \(\beta_j\) = expected change in \(Y\) per unit increase in \(X_j\)</td>
</tr>
<tr class="even">
<td><strong>Logistic Regression</strong></td>
<td>Binary \(Y \in \{0,1\}\)</td>
<td>Bernoulli</td>
<td>Wald tests <br> likelihood ratio tests <br> confidence intervals <br> odds ratio for one-unit increase in \(X_j\)</td>
</tr>
<tr class="odd">
<td><strong>Poisson Regression</strong></td>
<td>Count \(Y=0,1,2,\dots\)</td>
<td>Poisson</td>
<td>likelihood ratio tests <br> confidence intervals <br> multiplicative change in expected count for one-unit increase in \(X_j\)</td>
</tr>
<tr class="even">
<td><strong>Cox Regression</strong></td>
<td>Time-to-event (survival)</td>
<td>Partial likelihood</td>
<td>Standard errors from inverse information<br> score tests <br> confidence intervals<br>hazard ratio for one-unit increase in \(X_j\)</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="example-gene-expression-and-regression-analysis" class="slide level2">
<h2>Example: Gene Expression and regression analysis</h2>
<div style="font-size:0.7em">
<p>Gene expression varies across patients even when the underlying biological effect is stable.</p>
<p>We model expression as a random draw around an expected value:</p>
<p><span class="math display">\[
Y_i∣X_i∼N(X_iβ_i,σ2)
\]</span></p>
<ul>
<li><span class="math inline">\(X_iβ_i\)</span> : expected expression given covariates<br>
</li>
<li><span class="math inline">\(\sigma^2\)</span> : biological + technical variability<br>
</li>
<li>likelihood</li>
</ul>
<p><span class="math display">\[P(Y \mid X, \beta, \sigma^2)\]</span></p>
</div>
</section>
<section id="regression-modeling-questions" class="slide level2">
<h2>Regression modeling questions</h2>
<p>Once a likelihood is identified a fitted model can be used to find the following can be determined:</p>
<ol type="1">
<li><strong>How precise is my estimate of a drug effect?</strong><br>
</li>
<li><strong>Could the observed effect be due to chance?</strong><br>
</li>
<li><strong>What range of expression should I expect in a new patient?</strong></li>
</ol>
</section>
<section id="drug-effect-on-gene-expression" class="slide level2">
<h2>Drug effect on gene expression</h2>
<div style="font-size:0.7em">
<p>Simulated data:</p>
<ul>
<li>One gene</li>
<li>Two groups (Control vs Drug)</li>
<li>A stable effect with realistic variability</li>
</ul>
</div>
</section>
<section id="drug-effect-on-gene-expression-1" class="slide level2">
<h2>Drug effect on gene expression</h2>
<div style="font-size:0.7em">
<p>Simulated data:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb1-2"><a href=""></a></span>
<span id="cb1-3"><a href=""></a>n <span class="ot">&lt;-</span> <span class="dv">80</span></span>
<span id="cb1-4"><a href=""></a>drug_num <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">each =</span> n <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb1-5"><a href=""></a></span>
<span id="cb1-6"><a href=""></a>beta_0 <span class="ot">&lt;-</span> <span class="dv">5</span>      <span class="co"># baseline expression</span></span>
<span id="cb1-7"><a href=""></a>beta_1 <span class="ot">&lt;-</span> <span class="fl">1.2</span>    <span class="co"># drug effect (treated - control)</span></span>
<span id="cb1-8"><a href=""></a>sigma  <span class="ot">&lt;-</span> <span class="fl">1.0</span>    <span class="co"># biological variability</span></span>
<span id="cb1-9"><a href=""></a></span>
<span id="cb1-10"><a href=""></a>expression <span class="ot">&lt;-</span> beta_0 <span class="sc">+</span> beta_1 <span class="sc">*</span> drug_num <span class="sc">+</span> <span class="fu">rnorm</span>(n, <span class="dv">0</span>, sigma)</span>
<span id="cb1-11"><a href=""></a></span>
<span id="cb1-12"><a href=""></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb1-13"><a href=""></a>  <span class="at">expression =</span> expression,</span>
<span id="cb1-14"><a href=""></a>  <span class="at">drug =</span> <span class="fu">factor</span>(drug_num, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Drug"</span>))</span>
<span id="cb1-15"><a href=""></a>)</span>
<span id="cb1-16"><a href=""></a></span>
<span id="cb1-17"><a href=""></a><span class="fu">head</span>(dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  expression    drug
1   6.370958 Control
2   4.435302 Control
3   5.363128 Control
4   5.632863 Control
5   5.404268 Control
6   4.893875 Control</code></pre>
</div>
</div>
</div>
</section>
<section id="visualizing-biological-variability" class="slide level2">
<h2>Visualizing biological variability</h2>
<p>Expression values scatter around the expected mean in each group.</p>

<img data-src="test_2_082025_AIC_BIC_updated_files/figure-revealjs/unnamed-chunk-2-1.png" width="960" class="r-stretch"></section>
<section id="fit-the-regression-model" class="slide level2">
<h2>Fit the regression model</h2>
<div style="font-size:0.7em">
<p>Fit a linear moodel</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href=""></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(expression <span class="sc">~</span> drug, <span class="at">data =</span> dat)</span>
<span id="cb3-2"><a href=""></a><span class="fu">summary</span>(fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = expression ~ drug, data = dat)

Residuals:
     Min       1Q   Median       3Q      Max 
-3.07294 -0.60601  0.09052  0.64887  2.32618 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   4.9605     0.1708  29.045  &lt; 2e-16 ***
drugDrug      1.3194     0.2415   5.463 5.41e-07 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.08 on 78 degrees of freedom
Multiple R-squared:  0.2767,    Adjusted R-squared:  0.2674 
F-statistic: 29.84 on 1 and 78 DF,  p-value: 5.411e-07</code></pre>
</div>
</div>
</div>
</section>
<section id="fit-the-regression-model-1" class="slide level2">
<h2>Fit the regression model</h2>
<div style="font-size:0.7em">
<ul>
<li>primary result: drug effect on gene expression
<ul>
<li>treated samples show higher gene expression than controls [+1.32 expression units]</li>
</ul></li>
<li>estimate is tightly constrained by the data - standard error (drug effect): 0.24</li>
<li>observed effect of the drug on gene expression indicates true change
<ul>
<li>real treatment effect under the model’s probability assumptions: p-value: 5.4 × 10⁻⁷</li>
</ul></li>
</ul>
</div>
</section>
<section id="question-1-how-precise-is-my-estimate" class="slide level2">
<h2>Question 1: How precise is my estimate?</h2>
<div style="font-size:0.7em">
<ul>
<li>confidence interval summarizes precision for the drug effect</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href=""></a><span class="fu">confint</span>(fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                2.5 %   97.5 %
(Intercept) 4.6204621 5.300466
drugDrug    0.8385454 1.800216</code></pre>
</div>
</div>
<p>Interpretation for the drug coefficient:</p>
<ul>
<li>interval width reflects uncertainty driven by <span class="math display">\[\sigma^2\]</span> and sample size<br>
</li>
</ul>
</div>
</section>
<section id="question-2-could-the-effect-be-due-to-chance" class="slide level2">
<h2>Question 2: Could the effect be due to chance?</h2>
<div style="font-size:0.7em">
<ul>
<li>model evaluates compatibility with a null effect (drug = 0)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href=""></a><span class="fu">summary</span>(fit)<span class="sc">$</span>coefficients</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Estimate Std. Error   t value     Pr(&gt;|t|)
(Intercept) 4.960464  0.1707826 29.045488 1.402793e-43
drugDrug    1.319381  0.2415231  5.462753 5.411128e-07</code></pre>
</div>
</div>
<p>Interpretation for the drug treatment:</p>
<ul>
<li><strong>p-value</strong>: p-value: 5.4 × 10⁻⁷
<ul>
<li>computed from the sampling distribution implied by the Gaussian likelihood</li>
<li>less than 0.05 likely a true biological effect</li>
</ul></li>
</ul>
</div>
</section>
<section id="question-3-what-range-should-i-expect-in-a-new-patient" class="slide level2">
<h2>Question 3: What range should I expect in a new patient?</h2>
<div style="font-size:0.7em">
<ul>
<li>prediction interval describes plausible outcomes for an individual patient</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href=""></a>new_patient <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">drug =</span> <span class="fu">factor</span>(<span class="st">"Drug"</span>, <span class="at">levels =</span> <span class="fu">levels</span>(dat<span class="sc">$</span>drug)))</span>
<span id="cb9-2"><a href=""></a><span class="fu">predict</span>(fit, new_patient, <span class="at">interval =</span> <span class="st">"prediction"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       fit     lwr     upr
1 6.279845 4.10277 8.45692</code></pre>
</div>
</div>
<ul>
<li><span class="math inline">\(\hat{\mu}\)</span> (the estimated mean): estimate of gene expression change if a new patient were given the same treatment within a range specified by <span class="math inline">\(\sigma^2\)</span></li>
</ul>
</div>
</section>
<section id="model-comparisons" class="slide level2">
<h2>Model comparisons</h2>
<ul>
<li>multiple models are often run to generate estimates</li>
</ul>
<p>Model comparison formalizes questions like:</p>
<ul>
<li>Does adding predictors meaningfully improve our explanation?<br>
</li>
<li>Which model is best supported by the observed data?</li>
</ul>
</section>
<section id="what-counts-as-a-different-model" class="slide level2">
<h2>What counts as “a different model”?</h2>
<p>Two models differ when they change any of the following:</p>
<ul>
<li>Predictors included in the model</li>
<li>The distribution of the outcome (Gaussian vs Bernoulli)<br>
</li>
<li>The dependence structure (independent vs random effects)</li>
</ul>
</section>
<section id="fit-vs-complexity" class="slide level2">
<h2>Fit vs complexity</h2>
<ul>
<li><p>more complex models typically fit the observed data better</p></li>
<li><p>model comparison helps show whether:</p>
<ul>
<li>fit improves when the model captures real structure<br>
</li>
<li>complexity grows when the model has more parameters</li>
</ul></li>
<li><p>Model comparison methods reward fit and penalizes unnecessary complexity</p></li>
</ul>
</section>
<section id="two-common-model-comparison-situations" class="slide level2">
<h2>Two common model comparison situations</h2>
<p>In applied biomedical modeling, AIC and BIC are most often used in two recurring situations.</p>
<ul>
<li><p><strong>Incremental covariate adjustment (often nested models)</strong><br>
You evaluate whether adding a predictor (e.g., a biomarker, batch indicator, or clinical covariate) improves model support enough to justify the added complexity.</p></li>
<li><p><strong>Competing scientific hypotheses (often non-nested models)</strong><br>
You compare alternative representations of the biology for the <em>same</em> outcome, such as a linear dose effect versus a nonlinear effect.</p></li>
</ul>
<p>AIC and BIC compare models on a common likelihood-based scale while explicitly accounting for the number of parameters.</p>
</section>
<section id="what-aic-and-bic-are-optimizing" class="slide level2">
<h2>What AIC and BIC are optimizing</h2>
<p>AIC and BIC combine:</p>
<ul>
<li><strong>Fit:</strong> how well the model explains the observed data (via the maximized likelihood)</li>
<li><strong>Penalty:</strong> a cost for estimating more parameters</li>
</ul>
<p>They answer the question:</p>
<blockquote>
<p>“Did the improvement in likelihood justify the extra degrees of freedom?”</p>
</blockquote>
</section>
<section id="definitions" class="slide level2">
<h2>Definitions</h2>
<p>Let:</p>
<ul>
<li><span class="math display">\[\ell(\hat\\theta)\]</span> be the maximized log-likelihood under the fitted model</li>
<li><span class="math display">\[k\]</span> be the number of estimated parameters (including the intercept and <span class="math display">\[\\sigma^2\]</span> for Gaussian models)</li>
<li><span class="math display">\[n\]</span> be the sample size</li>
</ul>
<p>Then:</p>
<p><span class="math display">\[\\mathrm{AIC} = -2\\ell(\\hat\\theta) + 2k\]</span></p>
<p><span class="math display">\[\\mathrm{BIC} = -2\\ell(\\hat\\theta) + k\\log(n)\]</span></p>
<p>Lower is better for both criteria.</p>
</section>
<section id="how-to-interpret-aic-and-bic-differences" class="slide level2">
<h2>How to interpret AIC and BIC differences</h2>
<p>AIC and BIC are most useful through <em>differences</em>.</p>
<p>For a set of candidate models, define:</p>
<p><span class="math display">\[\\Delta_i = \\mathrm{IC}_i - \\min_j \\mathrm{IC}_j\]</span></p>
<ul>
<li><span class="math display">\[\\Delta = 0\]</span> identifies the best model by that criterion</li>
<li><span class="math display">\[\\Delta \\in [0,2]\]</span> indicates similar empirical support</li>
<li><span class="math display">\[\\Delta \\in [4,7]\]</span> indicates meaningfully less support</li>
<li><span class="math display">\[\\Delta &gt; 10\]</span> indicates very little support relative to the best</li>
</ul>
<p>This interpretation is applied the same way for AIC and BIC.</p>
</section>
<section id="background-what-aic-means" class="slide level2">
<h2>Background: what AIC means</h2>
<p>AIC arises from an information-theoretic view of model quality.</p>
<p>AIC selects the model expected to be closest (in Kullback–Leibler divergence) to the unknown data-generating process among the candidates.</p>
<p>In practice, AIC tends to favor models with strong out-of-sample predictive performance, and it is comfortable choosing moderately more complex models when they improve fit.</p>
</section>
<section id="background-what-bic-means" class="slide level2">
<h2>Background: what BIC means</h2>
<p>BIC is motivated by Bayesian model comparison.</p>
<p>Under common regularity conditions, BIC approximates the log Bayes factor for comparing models under a particular prior structure.</p>
<p>Because the penalty is <span class="math display">\[k\\log(n)\]</span>, BIC increases the cost of complexity as sample size grows, and it tends to favor simpler models unless the added parameters improve likelihood substantially.</p>
</section>
<section id="when-aic-and-bic-agree-and-disagree" class="slide level2">
<h2>When AIC and BIC agree and disagree</h2>
<p>They often agree when:</p>
<ul>
<li>A real biological signal is strong and the more complex model improves likelihood substantially.</li>
</ul>
<p>They often disagree when:</p>
<ul>
<li>An added effect is modest (small improvement in likelihood), or</li>
<li>Sample size is large (BIC penalizes complexity more strongly).</li>
</ul>
<p>In biomedical terms:</p>
<ul>
<li><strong>AIC</strong> supports models that generalize well for prediction and risk scoring.</li>
<li><strong>BIC</strong> supports models that are strongly justified as a parsimonious scientific explanation.</li>
</ul>
</section>
<section id="practical-workflow-what-you-do-in-analysis" class="slide level2">
<h2>Practical workflow (what you do in analysis)</h2>
<ol type="1">
<li>Fit a small set of scientifically motivated candidate models.</li>
<li>Compute AIC and BIC for each.</li>
<li>Compare <span class="math display">\[\\Delta\\mathrm{AIC}\]</span> and <span class="math display">\[\\Delta\\mathrm{BIC}\]</span> to identify supported models.</li>
<li>Interpret what the selected model implies biologically (which effects are strongly supported).</li>
</ol>
</section>
<section id="simulated-biological-example" class="slide level2">
<h2>Simulated biological example</h2>
<p>We simulate a gene-expression outcome (e.g., log2 expression) measured in patients.</p>
<p>Scientific question:</p>
<ul>
<li>Does a continuous biomarker improve explanation of expression beyond treatment status alone?</li>
<li>Is there evidence that treatment effect depends on biomarker level (an interaction)?</li>
</ul>
</section>
<section id="simulate-data-gene-expression-treatment-biomarker" class="slide level2">
<h2>Simulate data (gene expression, treatment, biomarker)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href=""></a><span class="fu">set.seed</span>(<span class="dv">29</span>)</span>
<span id="cb11-2"><a href=""></a></span>
<span id="cb11-3"><a href=""></a>n <span class="ot">&lt;-</span> <span class="dv">120</span></span>
<span id="cb11-4"><a href=""></a>treat <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(n, <span class="dv">1</span>, <span class="fl">0.5</span>)                 <span class="co"># 0 = control, 1 = drug</span></span>
<span id="cb11-5"><a href=""></a>biomarker <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>)    <span class="co"># standardized biomarker score</span></span>
<span id="cb11-6"><a href=""></a></span>
<span id="cb11-7"><a href=""></a><span class="co"># Data-generating biology:</span></span>
<span id="cb11-8"><a href=""></a><span class="co"># - baseline expression</span></span>
<span id="cb11-9"><a href=""></a><span class="co"># - treatment shifts expression upward</span></span>
<span id="cb11-10"><a href=""></a><span class="co"># - biomarker is associated with expression</span></span>
<span id="cb11-11"><a href=""></a><span class="co"># - treatment works better when biomarker is higher (interaction)</span></span>
<span id="cb11-12"><a href=""></a>y <span class="ot">&lt;-</span> <span class="dv">10</span> <span class="sc">+</span></span>
<span id="cb11-13"><a href=""></a>  <span class="fl">0.8</span> <span class="sc">*</span> treat <span class="sc">+</span></span>
<span id="cb11-14"><a href=""></a>  <span class="fl">1.2</span> <span class="sc">*</span> biomarker <span class="sc">+</span></span>
<span id="cb11-15"><a href=""></a>  <span class="fl">0.6</span> <span class="sc">*</span> treat <span class="sc">*</span> biomarker <span class="sc">+</span></span>
<span id="cb11-16"><a href=""></a>  <span class="fu">rnorm</span>(n, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="fl">1.5</span>)</span>
<span id="cb11-17"><a href=""></a></span>
<span id="cb11-18"><a href=""></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(y, <span class="at">treat =</span> <span class="fu">factor</span>(treat, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Drug"</span>)), biomarker)</span>
<span id="cb11-19"><a href=""></a><span class="fu">head</span>(dat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          y   treat  biomarker
1  9.324421 Control -0.2590092
2 13.413836 Control  1.0352162
3  7.452651 Control -0.7519736
4  8.458286 Control  0.1145837
5 10.066736    Drug -1.4768713
6  9.903991 Control  1.3361794</code></pre>
</div>
</div>
</section>
<section id="fit-candidate-models" class="slide level2">
<h2>Fit candidate models</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href=""></a>m1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treat, <span class="at">data =</span> dat)</span>
<span id="cb13-2"><a href=""></a>m2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treat <span class="sc">+</span> biomarker, <span class="at">data =</span> dat)</span>
<span id="cb13-3"><a href=""></a>m3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> treat <span class="sc">*</span> biomarker, <span class="at">data =</span> dat)  <span class="co"># includes main effects + interaction</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="compute-aic-and-bic" class="slide level2">
<h2>Compute AIC and BIC</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href=""></a>ic <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-2"><a href=""></a>  <span class="at">model =</span> <span class="fu">c</span>(<span class="st">"m1: treat"</span>, <span class="st">"m2: treat + biomarker"</span>, <span class="st">"m3: treat * biomarker"</span>),</span>
<span id="cb14-3"><a href=""></a>  <span class="at">AIC  =</span> <span class="fu">c</span>(<span class="fu">AIC</span>(m1), <span class="fu">AIC</span>(m2), <span class="fu">AIC</span>(m3)),</span>
<span id="cb14-4"><a href=""></a>  <span class="at">BIC  =</span> <span class="fu">c</span>(<span class="fu">BIC</span>(m1), <span class="fu">BIC</span>(m2), <span class="fu">BIC</span>(m3))</span>
<span id="cb14-5"><a href=""></a>)</span>
<span id="cb14-6"><a href=""></a></span>
<span id="cb14-7"><a href=""></a>ic<span class="sc">$</span>DeltaAIC <span class="ot">&lt;-</span> ic<span class="sc">$</span>AIC <span class="sc">-</span> <span class="fu">min</span>(ic<span class="sc">$</span>AIC)</span>
<span id="cb14-8"><a href=""></a>ic<span class="sc">$</span>DeltaBIC <span class="ot">&lt;-</span> ic<span class="sc">$</span>BIC <span class="sc">-</span> <span class="fu">min</span>(ic<span class="sc">$</span>BIC)</span>
<span id="cb14-9"><a href=""></a></span>
<span id="cb14-10"><a href=""></a>ic[<span class="fu">order</span>(ic<span class="sc">$</span>AIC), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  model      AIC      BIC  DeltaAIC   DeltaBIC
3 m3: treat * biomarker 474.9195 488.8570   0.00000   1.677572
2 m2: treat + biomarker 476.0295 487.1794   1.10992   0.000000
1             m1: treat 579.4836 587.8461 104.56408 100.666674</code></pre>
</div>
</div>
</section>
<section id="key-results-to-report-aicbic" class="slide level2">
<h2>Key results to report (AIC/BIC)</h2>
<p>From the table:</p>
<ul>
<li>The <strong>lowest AIC</strong> identifies the model with the best expected predictive support among candidates.</li>
<li>The <strong>lowest BIC</strong> identifies the most strongly supported parsimonious explanation among candidates.</li>
<li><span class="math display">\[\\Delta\]</span> values quantify how much less supported the other models are relative to the best.</li>
</ul>
<p>In a typical run of this simulation:</p>
<ul>
<li>Models that include the biomarker (m2 or m3) are strongly preferred over treatment-only (m1).</li>
<li>AIC may prefer the interaction model (m3) when the interaction improves fit enough.</li>
<li>BIC may choose m2 if the interaction improvement is modest relative to its extra parameter.</li>
</ul>
</section>
<section id="biological-interpretation" class="slide level2">
<h2>Biological interpretation</h2>
<p>When AIC and BIC both move away from treatment-only (m1):</p>
<ul>
<li>The biomarker carries real signal about expression variation.</li>
<li>The treatment effect estimate is meaningfully refined by accounting for biomarker-driven heterogeneity.</li>
</ul>
<p>When AIC prefers the interaction model (m3):</p>
<ul>
<li>The data support effect modification: <strong>drug response depends on biomarker level</strong>.</li>
<li>Biologically, this matches a stratified response mechanism: patients with higher biomarker values show larger expression shifts.</li>
</ul>
<p>When BIC prefers the additive model (m2):</p>
<ul>
<li>The biomarker is supported as a main effect.</li>
<li>Evidence for interaction is weaker relative to the complexity cost, so the most defensible explanation is <strong>parallel drug effect across biomarker levels</strong>.</li>
</ul>
</section>
<section id="a-compact-reporting-sentence-what-goes-in-a-results-paragraph" class="slide level2">
<h2>A compact reporting sentence (what goes in a results paragraph)</h2>
<p>“We compared candidate linear models using AIC and BIC. Models including the biomarker were strongly supported relative to the treatment-only model (large <span class="math display">\[\\Delta\\mathrm{AIC}\]</span> and <span class="math display">\[\\Delta\\mathrm{BIC}\]</span>). AIC favored the interaction model when treatment response varied by biomarker level, whereas BIC more often selected the additive model, indicating that interaction evidence was weaker relative to the complexity penalty at this sample size.”</p>
</section>
<section id="takeaway-for-practice" class="slide level2">
<h2>Takeaway for practice</h2>
<p>AIC and BIC provide fast, reproducible model comparison that aligns with scientific decision-making:</p>
<ul>
<li>Use <strong>AIC</strong> when your goal is stable predictive performance across new patients.</li>
<li>Use <strong>BIC</strong> when your goal is a concise biological explanation with strong support.</li>
</ul>
<p>In both cases, the biology is interpreted through the selected terms: main effects indicate additive contributions; interactions indicate effect modification and stratified response patterns.</p>

</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="test_2_082025_AIC_BIC_updated_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"8\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>